### ISSUE TITLE
1.7.0 breaks linux compatibility with pyinstaller

### ISSUE THREAD

**Issue by lexelby** [Toblerity/Shapely#831]
## Expected behavior and actual behavior.

Expected: Shapely is able to load libgeos_c.so.1 in a pyinstaller-packaged application.
Actual: Program throws an exception on startup:

```
Traceback (most recent call last):
  File "/home/runner/work/inkstitch/inkstitch/inkstitch.py", line 8, in <module>
  File "/home/runner/.local/lib/python2.7/site-packages/PyInstaller/loader/pyimod03_importers.py", line 396, in load_module
  File "/home/runner/work/inkstitch/inkstitch/lib/extensions/__init__.py", line 1, in <module>
  File "/home/runner/.local/lib/python2.7/site-packages/PyInstaller/loader/pyimod03_importers.py", line 396, in load_module
  File "/home/runner/work/inkstitch/inkstitch/lib/extensions/auto_satin.py", line 5, in <module>
  File "/home/runner/.local/lib/python2.7/site-packages/PyInstaller/loader/pyimod03_importers.py", line 396, in load_module
  File "/home/runner/work/inkstitch/inkstitch/lib/i18n.py", line 6, in <module>
  File "/home/runner/.local/lib/python2.7/site-packages/PyInstaller/loader/pyimod03_importers.py", line 396, in load_module
  File "/home/runner/work/inkstitch/inkstitch/lib/utils/__init__.py", line 3, in <module>
  File "/home/runner/.local/lib/python2.7/site-packages/PyInstaller/loader/pyimod03_importers.py", line 396, in load_module
  File "/home/runner/work/inkstitch/inkstitch/lib/utils/geometry.py", line 3, in <module>
  File "/home/runner/.local/lib/python2.7/site-packages/PyInstaller/loader/pyimod03_importers.py", line 396, in load_module
  File "/home/runner/.local/lib/python2.7/site-packages/shapely/geometry/__init__.py", line 4, in <module>
  File "/home/runner/.local/lib/python2.7/site-packages/PyInstaller/loader/pyimod03_importers.py", line 396, in load_module
  File "/home/runner/.local/lib/python2.7/site-packages/shapely/geometry/base.py", line 18, in <module>
  File "/home/runner/.local/lib/python2.7/site-packages/PyInstaller/loader/pyimod03_importers.py", line 396, in load_module
  File "/home/runner/.local/lib/python2.7/site-packages/shapely/coords.py", line 8, in <module>
  File "/home/runner/.local/lib/python2.7/site-packages/PyInstaller/loader/pyimod03_importers.py", line 396, in load_module
  File "/home/runner/.local/lib/python2.7/site-packages/shapely/geos.py", line 196, in <module>
  File "/home/runner/.local/lib/python2.7/site-packages/shapely/geos.py", line 184, in _geos_version
AttributeError: 'NoneType' object has no attribute 'GEOSversion'
```

This seems to happen because `_lgeos` is `None`.  The logic in #485 doesn't do anything if the geos_c library file is named `libgeos_c.so.1` as happens when Shapely is installed NOT as a wheel.  It seems to me as if everything would work if the outer `else` clause were run, but it isn't.

When installing Shapely as a wheel, the problem goes away.  In case you're curious, we were installing Shapely with `--no-binary` to work around the very problem that #485 seems to fix.  The wheel has libgeos_c in a different spot, and pyinstaller couldn't find it.

## Steps to reproduce the problem.

Package a python application that uses Shapely v1.7.0 using pyinstaller 3.3.1.  Run the program.

## Operating system

```Linux 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux```

## Shapely version and provenance

Shapely 1.7.0 installed from pypi _not_ as a whl (with `pip install --no-binary`).

**Comment by sgillies** [comment]
We really need someone who is invested in pyinstaller to take the lead on this issue. I'm happy to review and merge a PR that makes shapely work reliably with pyinstaller.

**Comment by lexelby** [comment]
Fair enough.  I don't think I'm they one to do that since I don't know much about pyinstaller.  I do think I know enough about this particular bug to submit a PR, if you're interested?

**Comment by sgillies** [comment]
I think an important thing is to have a test to guard against regressions, which this is, right? I'm not sure how easy or hard this would be to do with pyinstaller.

**Comment by durden** [comment]
Guess we could build a simple 'app' using shapely then run that through pyinstaller with the CI?

To do this I think we'd just need a small script that imports shapely, does some trivial 'work' that's fast and triggers an import of any DLL/so files. Then, in the CI just install the latest version of pyinstaller and run something like `pyinstaller <script>` and finally run the resulting exe and verify it doesn't crash on launch with something like `ImportError`.

**Comment by durden** [comment]
Can you try building the exe with the latest version of PyInstaller to see if you have the same issue? I think when I made the original fix that was merged into 1.7.0 I was using PyInstaller 3.4.0. It's possible your older version of PyInstaller has a hook file that doesn't quite work.

**Comment by djhoese** [comment]
FYI I just ran in to this with Pyinstaller master and the most recent shapely from conda-forge. I don't think pyinstaller is always happy with conda-forge install in general but I'll see if I can come up with something. It would be easier in pyinstaller if the hook could just not support versions older than 1.7. All of these pull requests to fix things in different versions are starting to get hard to follow.

**Comment by sgillies** [comment]
I'm going to close this. The entire category of errors involving dynamic loading of GEOS is going away in 2.0 and we won't be doing anything about any of them in the meanwhile.
